backend/src/main/java/com/cruru/advice/GlobalExceptionHandler.java

```diff
@@ -1,76 +1,89 @@
+import com.cruru.global.util.MdcUtils;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.Objects;
-import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.MDC;
+import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
 import org.springframework.http.ProblemDetail;
 import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
-import org.springframework.web.context.request.RequestContextHolder;
-import org.springframework.web.context.request.ServletRequestAttributes;
 import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 
+@Slf4j
 @RestControllerAdvice
-@RequiredArgsConstructor
-public class GlobalExceptionHandler {
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
 
-    private final UncatchedExceptionHandler handler;
+    private static final int STACK_TRACE_LIMIT = 5;
 
     @ExceptionHandler(CruruCustomException.class)
-    public ResponseEntity<ProblemDetail> handleCustomException(CruruCustomException e) {
-        HttpServletRequest request = getCurrentHttpRequest();
-        ExceptionLogger.info(request, e);
-
+    public ResponseEntity<Object> handleCustomException(CruruCustomException e) {
         ProblemDetail problemDetail = ProblemDetail.forStatusAndDetail(e.getStatus(), e.getMessage());
-        return ResponseEntity.of(problemDetail).build();
+
+        return handleExceptionWithMdc(e, problemDetail,
+                ex -> log.info("CRURU-EXCEPTION [errorMessage = {}]", ex.getMessage())
+        );
     }
 
     @ExceptionHandler(Exception.class)
-    public ResponseEntity<ProblemDetail> handleUnexpectedException(Exception e, WebRequest request) {
-        ProblemDetail problemDetail = handleException(e, request);
-        HttpStatus statusCode = HttpStatus.valueOf(problemDetail.getStatus());
-        if (statusCode.is5xxServerError()) {
-            ExceptionLogger.error(problemDetail);
-        } else {
-            ExceptionLogger.warn(problemDetail);
-        }
-        ProblemDetail detailsToSend = ProblemDetail.forStatus(statusCode);
-        return ResponseEntity.of(detailsToSend).build();
+    public ResponseEntity<Object> handleUnexpectedException(Exception e) {
+        ProblemDetail problemDetail = ProblemDetail.forStatusAndDetail(HttpStatus.INTERNAL_SERVER_ERROR, "서버 오류입니다.");
+
+        return handleExceptionWithMdc(e, problemDetail,
+                ex -> log.error("SERVER-EXCEPTION [errorMessage = {}, stackTrace = {}]", ex.getMessage(),
+                        getLimitedStackTrace(ex))
```

Comment:
```suggestion
                ex -> log.error(
                        "SERVER-EXCEPTION [errorMessage = {}, stackTrace = {}]",
                        ex.getMessage(),
                        getLimitedStackTrace(ex)
                )
```
개행을 전체적으로 조금 수정하는 것이 가독성이 더 좋을 것 같아요~ 최소한 여기만이라도..